File: src/main/java/eotree/EOComment.java
package eotree;

/**
 * EBNF representation:
 * <code>
 * '#' ANY EOL
 * </code>
 */
public class EOComment extends EONode {
    String comment;

    public EOComment(String comment) {
        this.comment = comment;
    }

    @Override
    public String generateEO(int indent) {
        return "# " + comment;
    }
}

File: src/main/java/eotree/EOComment.kt
package eotree

/**
 * EBNF representation:
 * `
 * '#' ANY EOL
` *
 */
class EOComment(var comment: String) : EONode() {
    override fun generateEO(indent: Int): String =
        "# $comment"

    override fun toString(): String =
        "# $comment"
}